### PR TITLE
Migrate layout pass

### PR DESCRIPTION
**Pull request by PoignardAzur**

### CODE REVIEW

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:3]:
```suggestion

//! The layout pass, which provides the size and position of each widget
//! before any translations applied in [`compose`](super::compose).
//! Most of the logic for this pass happens in [`Widget::layout`] implementations,
//!
//! In the future, this might also support scaling the child widgets.

```

File: masonry/src/widget/widget_pod.rs
Comment (DJMcNab) [inline masonry/src/widget/widget_pod.rs:507]: Why is this still here?

File: masonry/src/contexts.rs
Comment (DJMcNab) [inline masonry/src/contexts.rs:868]: We should ensure that this method is clearly pointed to in the docs for `Widget::layout`.

I'd also encourage moving this method higher up in this impl block, because it is among the most important methods on `LayoutCtx`

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:170]: No, that's a typo. It should be `pub(crate)`.

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:3]: I think scaling the child widgets would happen in compose.

Comment (DJMcNab) [reply masonry/src/passes/layout.rs:3]: Maybe. It depends on what purpose the scaling has

Because for text, layout can depend on the absolute logical size of the text, due to the `opsz` (optical size) variable font axis. So it depends on whether the scaling is logically scaling the logic size

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs]: Fixed.

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:3]: Added, except for the part about scaling children.

File: masonry/src/widget/widget_pod.rs
Comment (PoignardAzur) [reply masonry/src/widget/widget_pod.rs:436]: To keep the diff small. We can change all the widgets in a follow-up PR. I'll include a comment to that effect.

Comment (PoignardAzur) [reply masonry/src/widget/widget_pod.rs:433]: Done.

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:72]: Would this be the case to evaluate a potentially cached layout result? I guess that would be what `request_layout` would be needed for?

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:81]: I'm not a *huge* fan of using `Add` for `Insets`, but that's a Kurbo thing - this is an idiomatic usage thereof

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:142]: Should we log the id here as well?

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:149]: We should have a small doc comment here. E.g.:
```suggestion
/// Run [`Widget::layout`] method on the widget contained in `pod`.
/// This will be called by [`LayoutCtx::run_layout`], which is itself called in the parent widget's `layout`.
pub(crate) fn run_layout_on<W: Widget>(
```

File: masonry/src/widget/widget_pod.rs
Comment (DJMcNab) [reply masonry/src/widget/widget_pod.rs:433]: All of the widgets already appear to be translated in this PR?

**Review by DJMcNab — APPROVED** [review]

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:29]: I think this pattern pulled more weight when every WidgetPod method used it and `call_widget_method_with_checks` was used everywhere. We'll probably refactor it away soon. In the meantime, I added a comment for clarity.

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:55]: I think I wrote this code before I renamed those flags. Will fix.

File: masonry/src/passes/layout.rs
Comment (PoignardAzur) [reply masonry/src/passes/layout.rs:73]: True. I'm adding a comment and leaving it for a future PR.

**Comment by PoignardAzur** [comment]
Will merge before this evening unless a blocker comes up.

File: masonry/src/passes/layout.rs
Comment (DJMcNab) [inline masonry/src/passes/layout.rs:89]: There's a little bit of knowledge encoded in this line, so it could be worth having a comment here.

Not going to block on that, though

**Pull request merged**
